File: fizzbuzz.clj
(defn fizzbuzz [n]
  (let [divisible-by? (fn [div num] (= 0 (mod num div)))]
    (condp divisible-by? %
      15 "FizzBuzz"
      3 "Fizz"
      5 "Buzz"
      %)))

(defn fizz-n-and-a-buzz-n
  "Read procedure name using your Cosby voice."
  [n]
  (apply println (map fizzbuzz (range 1 (inc n)))))

;; It works. Trust me. It works!!!

File: niftyn8+rust/fizzbuzz.rs
fn divisible_by(num: int, divider: int) -> bool {
    num % divider == 0
}

fn main() {
    for num in range(1, 101) {
        println!("{}",
                 if divisible_by(num, 15) { "FizzBuzz".to_str() }
                 else if divisible_by(num, 3) { "Fizz".to_str() }
                 else if divisible_by(num, 5) { "Buzz".to_str() }
                 else { num.to_str() }
                );
    }
}
